src/analyze_fs.c

```diff
@@ -351,12 +351,23 @@ static void iterate_directory(struct manifest *manifest, char *pathprefix,
 
 		string_or_die(&fullname, "%s/%s", fullpath, entry->d_name);
 		populate_file_struct(file, fullname);
-		free(fullname);
 
 		if (entry->d_type == DT_DIR) {
 			iterate_directory(manifest, pathprefix, file->filename, do_hash);
+		} else if (entry->d_type == DT_UNKNOWN) { /* fall back to lstat() */
+			struct stat sb;
```

Comment:
Well it's populating the 'file' struct with the lstat details. That being the case, wouldn't it be easier to just  call

if (S_ISDIR(file.st_mode)) { 

and remove the new call to lstat?